modules/passkey/src/utils/webauthn.ts

```diff
@@ -0,0 +1,186 @@
+import { ethers } from 'ethers'
+import type { BigNumberish, BytesLike } from 'ethers'
+import CBOR from 'cbor'
+
+/**
+ * Returns the flag for the user verification requirement.
+ *
+ * See: <https://w3c.github.io/webauthn/#enumdef-userverificationrequirement>
+ */
+export function userVerificationFlag(userVerification: UserVerificationRequirement = 'preferred'): number {
+  switch (userVerification) {
+    case 'preferred':
+      return 0x01
+    case 'required':
+      return 0x04
+    default:
+      throw new Error(`user verification requirement ${userVerification} not supported`)
+  }
+}
+
+/**
+ * Returns the message that gets signed by the WebAuthn credentials.
+ *
+ * See <https://w3c.github.io/webauthn/#fig-signature>
+ */
+export function encodeWebAuthnSigningMessage(
+  clientData: { type: 'webauthn.get'; challenge: string; [key: string]: unknown },
+  authenticatorData: BytesLike,
+) {
+  return ethers.getBytes(ethers.concat([authenticatorData, ethers.sha256(ethers.toUtf8Bytes(JSON.stringify(clientData)))]))
+}
```

Comment:
This is a testing function, since the signing message should only ever be computed by the `WebAuthn` implementation in the browser.

This can move to the `test/` folder and make the `src/` no longer depend on `sha256` hashing.

Reply:
Moved 👍🏾